aircraft/cost.py

```diff
@@ -0,0 +1,112 @@
+    propulsion, tooling or production planning.
+
+    Tooling hours covers all the preparation for production.  It also covers
+    ongoing tooling cost through production.
+
+    Manufacturing hours is the direct labor to fabricate the aircraft. It also
+    includes cost of labor of subcontractors.
+
+    Quality Control hours covers all inspections.
+
+    Flight test cost covers all cost to prove airworthiness.
+
+    Manufacturing Materials is the raw materials cost and assumes that airframe
+    and main materials are made of aluminum.
+
+    The Engine Cost assumes that the cost of the engine is not known.
+
+    Uses 2012 pricing and economics
+
+    Units are not included in variable definitions
+
+    Labor Rates are found from Raymer
+    Avionics rate taken from Raymer and based on 15% cost of unit
+    Default values based on Honda Jet specs
+
+    Arguments
+    ---------
+    engine_defined : bool
+        If true, must specify T_max, M_max, and T_tur
+        If false, must define engine_cost
+        Todo: this should really be replaced with an engine cost (sub) model.
+    """
+    def setup(self, W_e=9200, V=420, Q=5, FTA=1, N_eng=10, T_max=2050,
+              M_max=0.63, T_tur=2050, R_avn=5000, engine_cost=500000,
+              eng_defined=False):
+
+        # User Definied Variables
+        W_e = Var('W_e', W_e, "lb", "Empty Weight")
+        V = Var('V', V, "knots", "Maximum Velocity")
+        Q = Var('Q', Q, "-", "Number produced in 5 years")
+        FTA = Var('FTA', FTA, "-", "Number of Flight Test Aircraft")
+        N_eng = Var('N_{eng}', N_eng, "-",
+                    "Number of engines or Q*num of engines per aircraft")
+        T_max = Var('T_{max}', T_max, "lb", "Maximum Thrust")
+        M_max = Var('M_{max}', M_max, "-", "Maximum Mach # of engine")
+        T_tur = Var('T_{tur}', T_tur, "R", "Turbine Inlet Temperature")
+        R_avn = Var('C_{avn}', R_avn, "$/lb", "Avionics Cost")
+
+        # Constants Hourly Rates
+        R_E = Var('R_{E}', 115, "$/hr", "Enginering Hourly Rate")
+        R_T = Var('R_{T}', 118, "$/hr", "Tooling Hourly Rate")
+        R_M = Var('R_{M}', 108, "$/hr", "Manufacturing Hourly Rate")
+        R_Q = Var('R_{Q}', 98, "$/hr", "Quality Check Hourly Rate")
+
+        # Free Variables
+
+        # Hourly costs in hrs. Multiply by hourly work rate to get cost
+        H_E = Var('H_{E}', "hrs", "Engineering Time")
+        H_T = Var('H_{T}', "hrs", "Tooling Time")
+        H_M = Var('H_{M}', "hrs", "Manufacturing Time")
+        H_Q = Var('H_{Q}', "hrs", "Quality Check Time")
+
+        # Cost Varibles
+        C_D = Var('C_D', "$", "Development Cost")
+        C_F = Var('C_F', "$", "Flight Test Cost")
+        C_M = Var('C_M', "$", "Manufacturing Materials Cost")
+        C_fly = Var('C_{tot}', "$", "RDT&E Flyaway Cost")
+        C_avn = Var('C_{avn}', "$", "Cost of avionics")
+        C_max = Var('C_{max}', 1000000000000, "$", "Maximum Cost")
```

Comment:
I think we could reasonably delete the `C_max` variable and associated constraint -- this could be specified externally if someone wants it in their broader Model.